nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,79 @@
+
+const express = require("express");
+const router = express.Router();
+
+const meals = require("../data/meals.json");
+
+router.get("/", async (request, response) => {
+
+  try{
+  
+ let newMeals=meals;
+ const { obj, maxPrice, searchTitle, createdAfter, limit } = request.query;
+
+if (maxPrice || searchTitle || createdAfter || limit) {
+  
+  if (maxPrice) {
+    newMeals=newMeals.filter((meal) => meal.price <= parseInt(maxPrice));
+    response.json(newMeals);
```

Comment:
Here you are returning newMeals if there is a maxPrice query parameter. However, later in line 39 you are again returning newMeals, which results to an error. Every route, should only return once!
Therefore it makes sense to remove this and keep then next one, since the newMeals could be filtered more, if there is also a searchTitle query parameter for instance.